adapter/postgres/postgres.go

```diff
@@ -40,11 +38,19 @@ func Open(dsn string) (*Adapter, error) {
 				InsertDefaultValues: true,
 				ErrorFunc:           errorFunc,
 			},
+			DB: database,
 		},
 	}
-	adapter.DB, err = db.Open("postgres", dsn)
+}
+
+// Open postgres connection using dsn.
+func Open(dsn string) (_ *Adapter, err error) {
+	var database *db.DB
+	if database, err = db.Open("postgres", dsn); err != nil {
+		return nil, err
+	}
 
-	return adapter, err
+	return New(database), nil
```

Comment:
I think we can do it like this, since it's not that different?

```go
database, err = db.Open("postgres", dsn)
return New(database), err
```

This will also fix the coverage.